compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensionsOps.td

```diff
@@ -86,13 +86,45 @@ def MatchCastCompatibleTypesOp : Op<Transform_Dialect, "iree.match.cast_compatib
   let cppNamespace = "mlir::iree_compiler::IREE::transform_dialect";
 }
 
+def MatchDimBoundsOp : Op<Transform_Dialect, "iree.match.dim_bounds",
+    [IsolatedFromAbove,
+     MatchOpInterface,
+     SingleValueMatcher,
+     MemoryEffectsOpInterface]> {
+  let summary =
+      "Checks whether the size of a dim is within given bounds";
+  let description = [{
+    Checks whether a dim is within a specified lower and upper bound.
+
+    #### Return modes
+
+    Succeeds if the value's type is compatible with the target type, and
+    produces a silenceable failure otherwise. Produces a definite failure
+    if the operand is not associated with a single payload value.
+  }];
+
+  let arguments = (ins TransformValueHandleTypeInterface:$operand_handle,
+                       I64Attr:$dim,
+                       OptionalAttr<I64Attr>:$lower_bound,
+                       OptionalAttr<I64Attr>:$upper_bound);
+
+  // `<=` isn't a valid literal so use `le` instead.
+  let assemblyFormat = [{
+    ($lower_bound^ `le` )? $operand_handle `[` $dim `]`
+    (`le` $upper_bound^)?  attr-dict `:` type($operand_handle)
```

Comment:
What do you think about using `umin = X, umax = Y` like in the assume op? I think that's more consistent/readable

Reply:
If I put `umin, umax` next to each other, the auto-generated parser always assumes the first integerattr it parses is the lower bound, so I kept `umin = X, %0, umax = Y`